### PR TITLE
Journalist can create article

### PR DESCRIPTION
**Pull request by mauroavellaneda**
[PT](https://www.pivotaltracker.com/story/show/175137722)
## This PR contains:
- Acceptance tests
- Login package
- 'Create Article' button
- 'Create Articles' form
- Module 'createForm' to manage request
- Axios

### CODE REVIEW

**Review by ctl-lab — APPROVED** [review]

File: cypress/integration/journalistCanLoginAndSeeCreateButton.feature.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    });
+
+    it("Journalist can login", () => {
+      // cy.get('[data-cy="login"]').click();
+      cy.get('[data-cy="login-form"]').within(() => {
+        cy.get('[data-cy="email"]').type("journalist@mail.com");
+        cy.get('[data-cy="password"]').type("password");
+        cy.get('[data-cy="button"]').contains("Submit").click();
+      });
+      // cy.get('[data-cy="message"]').should("contain", "Welcome journalist@mail.com");
```
Comment (oliverochman) [inline cypress/integration/journalistCanLoginAndSeeCreateButton.feature.js]: remove commented out code

File: cypress/integration/journalistCanLoginAndSeeCreateButton.feature.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      cy.visit("/");
+    });
+    it("invalid credentials", () => {
+      // cy.get('[data-cy="login"]').click();
+      cy.get('[data-cy="login-form"]').within(() => {
+        cy.get('[data-cy="email"]').type("invalid@mail.com");
+        cy.get('[data-cy="password"]').type("wrong_password");
+        cy.get('[data-cy="button"]').contains("Submit").click();
+      });
+      // cy.get('[data-cy="message"]').should("contain", "Invalid credentials");
```
Comment (oliverochman) [inline cypress/integration/journalistCanLoginAndSeeCreateButton.feature.js]: this is a message you should display

File: src/components/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { Menu } from "semantic-ui-react";
+// import { useParams } from "react-router-dom";
+
+const Header = () => {
+  // const { role } = useParams();
```
Comment (oliverochman) [inline src/components/Header.jsx]: remove commented out code

File: src/modules/createForm.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+
+const createForm = async (props) => {
```
Comment (oliverochman) [inline src/modules/createForm.js]: go ahead with the same module structure we used on the user fronting client

Articles.create()

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: cypress/integration/journalistCanCreateArticle.feature.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    });
+    it("journalist can successfully create articles", () => {
+      cy.get('[data-cy="create-article"]').contains("Create Article").click();
+      cy.get('[data-cy="form-article"]').within(() => {
+        cy.get('[data-cy="title"]').type("Title");
+        cy.get('[data-cy="lead"]').type("Lead");
+        cy.get('[data-cy="category"]').click();
+        cy.get('[data-cy="category"]').contains("Politics").click();
+      });
+      cy.get('[data-cy="content"]').type("Content");
```
Comment (oliverochman) [inline cypress/integration/journalistCanCreateArticle.feature.js]: is the content input outside of the form?

File: cypress/integration/journalistCanLoginAndSeeCreateButton.feature.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      });
+      cy.visit("/");
+    });
+    
+    it("invalid credentials", () => {
+      cy.get('[data-cy="login-form"]').within(() => {
+        cy.get('[data-cy="email"]').type("invalid@mail.com");
+        cy.get('[data-cy="password"]').type("wrong_password");
+        cy.get('[data-cy="button"]').contains("Submit").click();
+      });
```
Comment (oliverochman) [inline cypress/integration/journalistCanLoginAndSeeCreateButton.feature.js:56]: please add a test to see error response

File: cypress/integration/journalistCanLoginAndSeeCreateButton.feature.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      cy.route({
+        method: "POST",
+        url: "http://localhost:3000/api/v1/auth/sign_in",
+        response: '{"message": "Invalid credentials"}',
+        status: "401",
+      });
+      cy.visit("/");
+    });
+    
+    it("invalid credentials", () => {
```
Comment (oliverochman) [inline cypress/integration/journalistCanLoginAndSeeCreateButton.feature.js:51]: what happens if a user that does not have the role of journalist log in?

File: src/modules/createForm.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import axios from "axios";
+
+const Articles = {
+  async createForm(props) {
+    let headers = JSON.parse(localStorage.getItem("J-tockAuth-Storage"));
+
+    try {
+      let result = await axios.post(
+        "/journalist/articles",
```
Comment (oliverochman) [inline src/modules/createForm.js]: do you have the journalist namespace on the backend?

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by SebastianN97 — APPROVED** [review]

File: src/App.jsx [hunk 9]
```diff
@@ -1,12 +1,31 @@
...
-}
+  const authenticated = useSelector((state) => state.authenticate);
+  const role = useSelector((state) => state.currentUser.role);
+
+  let display;
 
-export default App
+  if (authenticated && role === "journalist") {
+    display = <Redirect to="/journalist" />;
+  }
```
Comment (emtalen) [inline src/App.jsx:16]: Can you please an else here, what should happen if the logged in user is not a journalist? Can we redirect to a page that indicates for the registered user that he or she is in the wrong place?

**Review by emtalen — CHANGES_REQUESTED** [review]

**Pull request merged**